java-squid/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -0,0 +1,494 @@
+    if (!checkerDispatcher.executeCheckPreStatement(tree)) {
+      // Some of the check pre statement sink the execution on this node.
+      return;
+    }
+    switch (tree.kind()) {
+      case METHOD_INVOCATION:
+        MethodInvocationTree mit = (MethodInvocationTree) tree;
+        setSymbolicValueOnFields(mit);
+        // unstack arguments and method identifier
+        programState = ProgramState.unstack(programState, mit.arguments().size() + 1).a;
+        logState(mit);
+        programState = ProgramState.stackValue(programState, constraintManager.createSymbolicValue(mit));
+        break;
+      case LABELED_STATEMENT:
+      case SWITCH_STATEMENT:
+      case EXPRESSION_STATEMENT:
+      case PARENTHESIZED_EXPRESSION:
+        throw new IllegalStateException("Cannot appear in CFG: " + tree.kind().name());
+      case VARIABLE:
+        VariableTree variableTree = (VariableTree) tree;
+        ExpressionTree initializer = variableTree.initializer();
+        if (initializer == null) {
+          SymbolicValue sv = null;
+          if (terminator != null && terminator.is(Tree.Kind.FOR_EACH_STATEMENT)) {
+            sv = constraintManager.createSymbolicValue(variableTree);
+          } else if (variableTree.type().symbolType().is("boolean")) {
+            sv = SymbolicValue.FALSE_LITERAL;
+          } else if (!variableTree.type().symbolType().isPrimitive()) {
+            sv = SymbolicValue.NULL_LITERAL;
+          }
+          if (sv != null) {
+            programState = ProgramState.put(programState, variableTree.symbol(), sv);
+          }
+        } else {
+          Pair<ProgramState, List<SymbolicValue>> unstack = ProgramState.unstack(programState, 1);
+          programState = unstack.a;
+          programState = ProgramState.put(programState, variableTree.symbol(), unstack.b.get(0));
+        }
+        break;
+      case TYPE_CAST:
+        TypeCastTree typeCast = (TypeCastTree) tree;
+        Type type = typeCast.type().symbolType();
+        if (type.isPrimitive()) {
+          Pair<ProgramState, List<SymbolicValue>> unstack = ProgramState.unstack(programState, 1);
+          programState = unstack.a;
+          programState = ProgramState.stackValue(programState, constraintManager.createSymbolicValue(typeCast.expression()));
+        }
+        break;
+      case ASSIGNMENT:
+        ExpressionTree variable = ((AssignmentExpressionTree) tree).variable();
+        if (variable.is(Tree.Kind.IDENTIFIER)) {
+          // FIXME restricted to identifiers for now.
+          Pair<ProgramState, List<SymbolicValue>> unstack = ProgramState.unstack(programState, 2);
+          SymbolicValue value = unstack.b.get(1);
+          programState = unstack.a;
+          programState = ProgramState.put(programState, ((IdentifierTree) variable).symbol(), value);
+          programState = ProgramState.stackValue(programState, value);
+        }
+        break;
+      case ARRAY_ACCESS_EXPRESSION:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 6 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1151)